core/types/uuid.go

```diff
@@ -62,6 +87,134 @@ func (u *UUID) Scan(src any) error {
 	return errors.New("not a byte slice")
 }
 
+func (u *UUID) FormatSupported(format int16) bool {
+	return format == pgtype.TextFormatCode || format == pgtype.BinaryFormatCode
+}
+
+func (u *UUID) PreferredFormat() int16 {
+	return pgtype.TextFormatCode
```

Comment:
The UUID type built in to pgype prefers binary.
Will you document some of the decisions in this code and/or like to reference implementations?  There's a lot of logic add with no obvious reasons, although I'm sure there are indeed good ones.